nixos/modules/services/networking/syncthing.nix

```diff
@@ -151,6 +383,20 @@ in {
           RestartForceExitStatus="3 4";
           User = cfg.user;
           Group = cfg.group;
+          ExecStartPre = mkIf (cfg.declarative.cert != null || cfg.declarative.key != null) {
```

Comment:
```          ExecStartPre =
            let
              certScript = pkgs.writeShellScriptBin "certScript" (optionalString (cfg.declarative.cert != null)
              ''
                cp ${toString cfg.declarative.cert} ${cfg.configDir}/cert.pem
                chown ${cfg.user}:${cfg.group} ${cfg.configDir}/cert.pem
                chmod 400 ${cfg.configDir}/cert.pem
              '');
              keyScript = pkgs.writeShellScriptBin "keyScript" (optionalString (cfg.declarative.key != null)
              ''
                cp ${toString cfg.declarative.key} ${cfg.configDir}/key.pem
                chown ${cfg.user}:${cfg.group} ${cfg.configDir}/key.pem
                chmod 400 ${cfg.configDir}/key.pem
              '');
            in
            [
              "+${certScript}/bin/certScript"
              "+${keyScript}/bin/keyScript"
            ];
```
Fixing the issue I described, and `ExecStartPre` will be executed with super user rights.